### PR TITLE
feat: Add log file to each node with detailed information

### PR DESCRIPTION
**Pull request by alenmestrov**
**Issue Reference**  
Closes [#564](https://github.com/calimero-network/core/issues/564)

**Summary**  
This PR adds a new section to the Tauri-based desktop application, allowing users to view node logs in real-time. The logs capture the node's terminal output, initialization events, and stop notifications. Additionally, the backend now manages a log file for each node, with a size limit of 5MB. Once the log file exceeds this limit, older entries are deleted to maintain the file size.

**Changes**  
- **UI Enhancement**: Added a new "Node Logs" section to the frontend, where users can view the contents of the log file in real time.
- **Log File Creation**: Implemented logic in the backend to create a log file when a node is initialized.
- **Log Writing**: Captured and logged node terminal output on start, track node activity while active, and logged a stop notice when the node is stopped.
- **Log Size Management**: Introduced a 5MB size limit for each node's log file. When the file exceeds this limit, the oldest lines are deleted to maintain the file size.
- **Real-time Log Display**: Connected the frontend log display to the backend to show log content as it is written in real time.

**Testing**  
- Verified that a log file was created for each node upon initialization.
- Tested that terminal output is correctly written to the log file while the node is running and that a stop notice is logged when the node is stopped.
- Checked the log file size management, ensuring that once the 5MB limit is reached, older log entries are removed.
- Confirmed that the log content is displayed in the UI and updates in real-time as new entries are written.

**How to Test**  
1. Run the following commands:
   ```bash
   pnpm install
   pnpm run tauri dev
   ```
2. Initialize a node and check that a log file is created in the node's directory.
3. Start the node and verify that terminal output is logged both in the file and displayed in the "Node Logs" section of the UI.
4. Stop the node and check that a stop notice is added to the log.
5. Test the 5MB size limit by running the node with significant activity and ensuring older log lines are deleted once the limit is reached.

### CODE REVIEW

File: src-tauri/src/logger/logger.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            let bytes_read = reader.read_line(&mut line)?;
+            current_pos += bytes_read as u64;
+            line.clear();
+        }
+
+        // Truncate the file and write the remaining content
+        writer.set_len(0)?;
+        writer.seek(SeekFrom::Start(0))?;
+        io::copy(&mut reader, &mut writer)?;
+    }
```
Comment (miraclx) [inline src-tauri/src/logger/logger.rs:67]: not sure I understand this bit, we're reading from the same file we're writing to?

File: src-tauri/src/logger/mod.rs [hunk 2]
```diff
@@ -0,0 +1 @@
+pub mod logger;
```
Comment (miraclx) [inline src-tauri/src/logger/mod.rs:1]: why not inline the contents of logger?

File: src-tauri/src/logger/logger.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+use crate::utils::utils::get_nodes_dir;
+use std::fs::{self, File, OpenOptions};
+use std::io::{self, BufRead, BufReader, Seek, SeekFrom, Write};
+use std::path::{Path, PathBuf};
+
+use tauri::{AppHandle, State};
+
+const MAX_LOG_SIZE: usize = 5 * 1024 * 1024; // 5MB
+
+pub fn create_log_file(app_handle: &AppHandle, node_name: &str) -> Result<()> {
```
Comment (miraclx) [inline src-tauri/src/logger/logger.rs:12]: we can return `File` here

File: src-tauri/src/operations/node_operations.rs [hunk 4]
```diff
@@ -52,6 +54,21 @@ pub async fn create_node(
         });
     }
 
+    create_log_file(&state.app_handle, &node_name)?;
```
Comment (miraclx) [inline src-tauri/src/operations/node_operations.rs:57]: the check for file existence can happen here

File: src/components/node/NodeLogs.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import { LogsContainer, LogsOutput, LogsHeader } from '../../styles/NodeLogsStyles';
+import { NodeDetails } from '../../hooks/useNodeManagement';
+
+interface NodeLogsProps {
+  selectedNode: NodeDetails;
+  onClose: () => void;
+}
+
+const NodeLogs: React.FC<NodeLogsProps> = ({ selectedNode }) => {
+  const [logs, setLogs] = useState<string>('');
```
Comment (frdomovic) [inline src/components/node/NodeLogs.tsx:12]: so in some cases you use <string | null>(null) and here its (''). Any reason for this ? 
Think we can use just <string> in other cases also as "" will return false same as null when tested

**Review by frdomovic — APPROVED** [review]
LGTM, check comments

File: src-tauri/src/logger/logger.rs (hunk 1 above)
Comment (alenmestrov) [reply src-tauri/src/logger/logger.rs]: This is to limit the size of the log file which is currently set to 5MB.

File: src/components/node/NodeLogs.tsx (hunk 5 above)
Comment (alenmestrov) [reply src/components/node/NodeLogs.tsx]: updated, switched to use only strings

File: src-tauri/src/logger/mod.rs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        let mut reader = BufReader::new(file.try_clone()?); // Clone the file for reading
+        let writer = file; // Use the original file for writing
+                           // Skip lines until we're within the size limit
+        let mut current_pos = 0;
+        let mut line = String::new();
+        while current_pos < metadata.len() - MAX_LOG_SIZE as u64 {
+            let bytes_read = reader.read_line(&mut line)?;
+            current_pos += bytes_read as u64;
+            line.clear();
+        }
```
Comment (miraclx) [inline src-tauri/src/logger/mod.rs:53]: Can we seek by N bytes where N is how much over the max we're at?

```rust
reader.seek(SeekFrom::Start(metadata.len() - MAX_LOG_SIZE as u64));
```

which could potentially drop us in-between a line, so we ignore the first line we see

```rust
reader.read_line(&mut Default::default());
```

**Pull request merged**